### PR TITLE
App backup support

### PR DESCRIPTION
**Pull request by Rohit-PX**
Why do we need this PR?

Adding support for backup objects in torpedo.
* BackupLocation
* ApplicationBackup
* ApplicationRestore

### CODE REVIEW

File: drivers/scheduler/k8s/k8s.go [hunk 1]
```diff
@@ -856,6 +879,33 @@ func (k *k8s) WaitForRunning(ctx *scheduler.Context, timeout, retryInterval time
...
+					Cause: fmt.Sprintf("Failed to validate ApplicationBackup: %v. Err: %v", obj.Name, err),
+					Type:  obj,
+				}
+			}
+			logrus.Infof("[%v] Validated ApplicationBackup: %v", ctx.App.Key, obj.Name)
+		} else if obj, ok := spec.(*stork_api.ApplicationRestore); ok {
+			if err := k8sOps.ValidateApplicationRestore(obj.Name, obj.Namespace, timeout, retryInterval); err != nil {
+				return &scheduler.ErrFailedToValidateCustomSpec{
+					Name:  obj.Name,
+					Cause: fmt.Sprintf("Failed to validate ApplicationBackup: %v. Err: %v", obj.Name, err),
```
Comment (disrani-px) [inline drivers/scheduler/k8s/k8s.go]: `ApplicationRestore`

File: drivers/scheduler/k8s/k8s.go [hunk 2]
```diff
@@ -856,6 +879,33 @@ func (k *k8s) WaitForRunning(ctx *scheduler.Context, timeout, retryInterval time
...
+					Type:  obj,
+				}
+			}
+			logrus.Infof("[%v] Validated ApplicationBackupRestore: %v", ctx.App.Key, obj.Name)
```
Comment (disrani-px) [inline drivers/scheduler/k8s/k8s.go]: The type is ` ApplicationRestore`

File: drivers/scheduler/k8s/k8s.go (hunk 1 above)
Comment (Rohit-PX) [reply drivers/scheduler/k8s/k8s.go]: Fixed.

File: drivers/scheduler/k8s/k8s.go (hunk 2 above)
Comment (Rohit-PX) [reply drivers/scheduler/k8s/k8s.go]: Fixed.

**Pull request merged**